ding/entry/cli_parsers/slurm_parser.py

```diff
@@ -0,0 +1,96 @@
+import os
+import re
+
+
+class SlurmParser():
+
+    def __init__(self, platform_spec, **kwargs) -> None:
+        """
+        Overview:
+            Should only set global cluster properties
+        """
+        self.kwargs = kwargs
+        self.ntasks = int(os.environ["SLURM_NTASKS"])
+        self.tasks = platform_spec["tasks"]
+        self.ntasks_per_node = int(os.environ["SLURM_NTASKS_PER_NODE"])
+        self.nodelist = self._parse_node_list()
+
+    def parse(self):
+        assert len(self.tasks) == self.ntasks
+        procid = int(os.environ["SLURM_PROCID"])
+        nodename = os.environ["SLURMD_NODENAME"]
+        task = self._get_node_args(procid)
+        # Validation
+        assert task["address"] == nodename
+        return {**self.kwargs, **task}
```

Comment:
deepcopy self.kwargs for security

Reply:
this expression will not overwrite properties of kwargs, so there is no need to do deepcopy here.